src/analyses/ai.h

```diff
@@ -8,6 +8,39 @@ Author: Daniel Kroening, kroening@kroening.com
 
 /// \file
 /// Abstract Interpretation
+///
+/// This is the core of the abstract interpretation framework.
+/// To run an analysis you need four components:
+///
+///  1. An abstract interpreter, derived from ai_baset.
+///     This performs that actual analysis.  There are a number of alternative
+///     to choose from, primarily giving different ways of handling function
+///     calls / interprocedural analysis.
+///     More information is given in this file.
+///
+///  2. A history factory, derived from ai_history_factory_baset.
+///     This generates history objects (derived from ai_history_baset) which
+///     control the number of steps that the analysis performs.  These can be
+///     simple, just tracking location, or they can be complex, tracking
+///     location, function calls, branches (forwards and backwards), even
+///     threading.
+///     See ai_history.h for more information.
+///
+///  3. A domain factory, derived from ai_domain_factory_baset.
+///     This generates domain objects (derived from ai_domain_baset) which
```

Comment:
We should decide on a terminology here and then use it consistently to avoid confusion. Domain, domain object, domain element, domain value, abstract value, ... all of these are used somewhere (all of which are legit - except 'domain').

Reply:
Can that be an action for another PR as I think that issue is wider than just the code touched by this PR?

Reply:
I will as part of my "things to do once everyone is happy with this PR".